design/thrift-proxy-design.md

```diff
@@ -0,0 +1,215 @@
+# Thrift Proxy Design
+
+# Abstract
+
+Thrift Proxy is an L7 Envoy proxy that is used for routing Thrift protocol. 
+This document outlines the rationale, objectives, architectural design, and technical details of integrating Thrift Proxy with Contour. 
+The integration of Thrift Proxy into Contour opens up new possibilities for routing and managing Thrift-based applications.
+
+
+# Background
```

Comment:
A useful link I found looking into envoy + Thrift Proxy history, the original proposal to add Thrift support to Envoy: https://docs.google.com/document/d/1QuiZMT1ruetavVHsj4Sv55YpBFVxTEzJj-ZpOJqAboc/edit